splice_ck/src/main/java/com/splicemachine/ck/HBaseInspector.java

```diff
@@ -0,0 +1,176 @@
+import com.splicemachine.si.constants.SIConstants;
+import org.apache.hadoop.conf.Configuration;
+import org.apache.hadoop.hbase.*;
+import org.apache.hadoop.hbase.client.*;
+import org.apache.hadoop.hbase.util.Bytes;
+import org.apache.hadoop.hbase.util.Pair;
+
+import java.util.List;
+import java.util.regex.Pattern;
+
+public class HBaseInspector {
+
+    private Configuration config;
+
+    public HBaseInspector(Configuration config) {
+        this.config = config;
+    }
+
+    public static class ScanPrinter implements AutoCloseable {
+        @Override
+        public void close() {
+        }
+
+        public void ProcessRow(Result r, ICellVisitor visitor) throws Exception {
+            CellScanner scanner = r.cellScanner();
+            while (scanner.advance()) {
+                Cell cell = scanner.current();
+                visitor.visit(cell);
+            }
+        }
+    }
+
+    public String scanRow(String region, String rowKey, Utils.SQLType[] schema) throws Exception {
+        StringBuilder result = new StringBuilder();
+        try {
+            try (Connection conn = ConnectionFactory.createConnection(config)) {
+                Table table = conn.getTable(TableName.valueOf(region));
+                Scan scan = new Scan();
+                scan.withStartRow(Bytes.fromHex(rowKey)).setLimit(1).readAllVersions();
+                try (ResultScanner results = table.getScanner(scan)) {
+                    ScanPrinter scanner = new ScanPrinter();
+                    CellPrinter cellPrinter = new CellPrinter(new UserDefinedDataDecoder(schema, 4));
+                    for (Result r : results) {
+                        scanner.ProcessRow(r, cellPrinter);
+                        result.append(cellPrinter.getOutput());
+                    }
+                }
+            }
+        } catch (Exception e) {
+            return Utils.checkException(e, region);
+        }
+        return result.toString();
+    }
+
+    private Utils.Tabular getListTables() throws Exception {
+        try (Connection conn = ConnectionFactory.createConnection(config)) {
+            List<TableDescriptor> descriptors = conn.getAdmin().listTableDescriptors(Pattern.compile("splice.*"));
+
+            Utils.Tabular tabular = new Utils.Tabular("hbase name", "schema", "table", "index", "create txn");
+            for (TableDescriptor td : descriptors) {
+                tabular.addRow(CheckNull(td.getTableName().toString()),
+                        CheckNull(td.getValue(SIConstants.SCHEMA_DISPLAY_NAME_ATTR)),
+                        CheckNull(td.getValue(SIConstants.TABLE_DISPLAY_NAME_ATTR)),
+                        CheckNull(td.getValue(SIConstants.INDEX_DISPLAY_NAME_ATTR)),
+                        CheckNull(td.getValue(SIConstants.TRANSACTION_ID_ATTR)));
+            }
+            return tabular;
+        }
+    }
+
+    private String CheckNull(String value) {
+        return value == null ? "NULL" : value;
```

Comment:
maybe "\"" + value + "\"", to differentiate the string "NULL" from NULL ?
And maybe also escape strings? Not sure how "pro" this needs to be

```
if( value == null )
  return "NULL";

return "\"" + value.replace("\\", "\\\\").replace("\"", "\\\"") + "\"";
```

Reply:
that could get more complicated for its purpose, also we could easily make things more confusing if we tamper with the original values therefore I wouldn't do any escaping myself. Instead maybe we can give the user the option to tell us how nulls should look like. I would rather leave it as is at the moment.